cluster_management/src/main/thrift/events_history.thrift

```diff
@@ -0,0 +1,95 @@
+# Copyright 2016 Pinterest Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+namespace java com.pinterest.rocksplicator.thrift.eventhistory
+
+enum LeaderEventType {
+  # All of the event types PREFIXED with PARTICIPANT
+  # originates from PARTICIPANT which is an existing leader
+  # for a given partition or a PARTICIPANT which is trying
+  # to become a new leader.
+  PARTICIPANT_LEADER_DOWN_INIT = 1,
+  PARTICIPANT_LEADER_DOWN_SUCCESS = 2,
+  PARTICIPANT_LEADER_DOWN_FAILURE = 3,
+  PARTICIPANT_LEADER_UP_INIT = 4,
+  PARTICIPANT_LEADER_UP_SUCCESS = 5,
+  PARTICIPANT_LEADER_UP_FAILURE = 6,
+  # Following events are currently not used
+  # Placeholder events have been placed to
+  # help with future migration to new event types
+  PARTICIPANT_PLACEHOLDER_EVENT_TYPE_07 = 7
+  PARTICIPANT_PLACEHOLDER_EVENT_TYPE_08 = 8
+  PARTICIPANT_PLACEHOLDER_EVENT_TYPE_09 = 9
+  PARTICIPANT_PLACEHOLDER_EVENT_TYPE_10 = 10
+  PARTICIPANT_PLACEHOLDER_EVENT_TYPE_11 = 11
+  PARTICIPANT_PLACEHOLDER_EVENT_TYPE_12 = 12
+  PARTICIPANT_PLACEHOLDER_EVENT_TYPE_13 = 13
+  PARTICIPANT_PLACEHOLDER_EVENT_TYPE_14 = 14
+  PARTICIPANT_PLACEHOLDER_EVENT_TYPE_15 = 15
+  PARTICIPANT_PLACEHOLDER_EVENT_TYPE_16 = 16
+  PARTICIPANT_PLACEHOLDER_EVENT_TYPE_17 = 17
+  PARTICIPANT_PLACEHOLDER_EVENT_TYPE_18 = 18
+  PARTICIPANT_PLACEHOLDER_EVENT_TYPE_19 = 19
+  PARTICIPANT_PLACEHOLDER_EVENT_TYPE_20 = 20
```

Comment:
Enum Placeholders are good way to avoid issues with Enum deserialization when deploy goes out with new enum values. It avoids deploy dance... and save a lot of time in rolling out new enum Types...

Reply:
I'm curious to understand how this works. If we are adding new fields, they would not be used previously. But since thrift passes only the values, it should be backward compatible already. So I dont fully understand the need for placeholders.